File: core/transactions/distributed_transaction_coordinator.py
from __future__ import annotations

from typing import Any
from typing import Dict
from typing import List


def coordinate_transactions(
    transactions: List[
        Dict[str, Any]
    ],
) -> Dict[str, Any]:

    committed = []

    for tx in transactions:

        committed.append({
            "id": tx.get("id"),
            "committed": True,
        })

    return {
        "transactions": committed,
    }

File: src/transactions/distributedTransactionCoordinator.ts
/**
 * Converted from Python: core/transactions/distributed_transaction_coordinator.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function coordinateTransactions(transactions: any): any {
  var committed: any[] = [];
  var tx: any;
  for (tx of py.iter(transactions)) {
    py.listAppend(committed, {"id": py.get(tx, "id"), "committed": true});
  }
  return {"transactions": committed};
}
